zephyr/include/rtos/wait.h

```diff
@@ -24,9 +24,9 @@ static inline void idelay(int n)
 
 static inline void wait_delay(uint64_t number_of_clks)
 {
-	uint64_t timeout = k_cycle_get_64() + number_of_clks;
+	uint64_t timeout = sof_cycle_get_64() + number_of_clks;
```

Comment:
We should not be using the sof_ abstraction here as we ```wait_delay``` should call the native Zephyr API. If needed it would be better to call ```k_cycle_get_32()``` for i.MX on Zephyr here with an ifdef